docs/hub-tutorials/join-testnet.md

```diff
@@ -4,64 +4,159 @@ order: 4
 
-These instructions are for full nodes that have ran on previous versions of and would like to upgrade to the latest testnet.
+Cosmovisor requires the creation the following directory structure:
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── gaiad
+└── upgrades
+    └── Vega
+        ├── bin
+        │   └── gaiad
+        └── upgrade-info.json
+```
 
-### Reset Data
+It is possible to enable autodownload for the new binary, but for the purpose of this tutorial, the setup instructions will include how to do this manually. For more information on autodownload with Cosmovisor, see the Vega Testnet respository's [documentation on Cosmosvisor](https://github.com/cosmos/vega-test/blob/master/local-testnet/README.md#Cosmovisor).
 
-First, remove the outdated files and reset the data.
+Full instructions for using Cosmovisor with `gaiad` can be found under the Cosmos Hub [Upgrade tutorial](https://hub.cosmos.network/main/hub-tutorials/upgrade-node.html).
+
+> **Note**: Cosmos Hub recommends running `gaiad` or `cosmovisor` with the `--x-crisis-skip-assert-invariants` flag. If checking for invariants, operators are likely to see `rounding error withdrawing rewards from validator`. These are expected.
 
-```bash
-rm $HOME/.gaia/config/addrbook.json $HOME/.gaia/config/genesis.json
-gaiad unsafe-reset-all
 ```
+# Install Cosmovisor
+go get github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor
 
-Your node is now in a pristine state while keeping the original `priv_validator.json` and `config.toml`. If you had any sentry nodes or full nodes setup before,
-your node will still try to connect to them, but may fail if they haven't also
-been upgraded.
+# Set environment variables
+echo "export DAEMON_NAME=gaiad" >> ~/.profile
+echo "export DAEMON_HOME=$HOME/.gaia" >> ~/.profile
+source ~/.profile
 
-::: danger Warning
-Make sure that every node has a unique `priv_validator.json`. Do not copy the `priv_validator.json` from an old node to multiple new nodes. Running two nodes with the same `priv_validator.json` will cause you to double sign.
-:::
+mkdir -p ~/.gaia/cosmovisor/upgrades
+mkdir -p ~/.gaia/cosmovisor/genesis/bin/
+cp $(which gaiad) ~/.gaia/cosmovisor/genesis/bin/
+
+# Verify cosmovisor and gaiad versions are the same.
+cosmovisor version
+
+# Start Cosmovisor
+cosmovisor start
+```
 
-### Software Upgrade
+#### Upgrading
 
-Now it is time to upgrade the software:
+Cosmovisor will continually poll the `$DAEMON_HOME/data/upgrade-info.json` for new upgrade instructions. When an upgrade is ready, node operators can download the new binary and place it under `$DAEMON_HOME/cosmovisor/upgrades/<name>/bin` where `<name>` is the URI-encoded name of the upgrade as specified in the upgrade module plan.
 
-```bash
-git clone https://github.com/cosmos/gaia.git
-cd gaia
-git fetch --all && git checkout master
+For the `Vega` upgrade, this would look like:
```

Comment:
```suggestion
When the chain reaches block height `7,453,750`, the chain will halt and you will have to download the new binary and move it to the correct folder. For the `Vega` upgrade, this would look like:
```